pgmpy/tests/test_models/test_MarkovModel.py

```diff
@@ -216,12 +216,12 @@ def test_factor_graph(self):
 
         factor_graph = self.graph.to_factor_graph()
         self.assertIsInstance(factor_graph, FactorGraph)
-        self.assertListEqual(sorted(factor_graph.nodes()),
-                             ['Alice', 'Bob', 'Charles', 'phi_Alice_Bob',
-                              'phi_Bob_Charles'])
-        self.assertListEqual(hf.recursive_sorted(factor_graph.edges()),
-                             [['Alice', 'phi_Alice_Bob'], ['Bob', 'phi_Alice_Bob'],
-                              ['Bob', 'phi_Bob_Charles'], ['Charles', 'phi_Bob_Charles']])
+        self.assertSetEqual(set(factor_graph.nodes()),
+                            set(['Alice', 'Bob', 'Charles', phi1,
+                              phi2]))
+        self.assertSetEqual(set(factor_graph.edges()),
+                            set([('Alice', phi1), ('Bob', phi1),
```

Comment:
@VSSILPA Correct order would be:
`set([('Bob', phi1), ('Bob', phi2), ('Charles', phi2), (phi1, 'Alice')])`

You can either change it to above or you can convert the tuples to frozenset in both cases. I'd like to go with latter one, add this function in help_function.py in test directory.

Reply:
I've converted tuples to frozenset. Is this fine?

Reply:
It's working for all test cases now.